src/Catel.Core/Catel.Core.Shared/Data/ModelBase.propertyediting.cs

```diff
@@ -122,7 +122,7 @@ internal void SetValue(PropertyData property, object value, bool notifyOnChange,
                 }
 
                 // Validate before assigning, dynamic properties will cause exception
-                if (validateAttributes && !LeanAndMeanModel)
+                if (validateAttributes && !LeanAndMeanModel && ValidateUsingDataAnnotations)
```

Comment:
Perhaps I thought to complex today: Just add the check within "ValidatePropertyUsingAnnotations" as mentioned. That's better in terms of calling the function from other locations. Furthermore there is no real need to pass through that parameter because it's already evaluated in "Validate()" :-)